packages/fast-components-styles-msft/src/utilities/color/neutral-fill-input.ts

```diff
@@ -0,0 +1,81 @@
+import {
+    DesignSystem,
+    DesignSystemResolver,
+    ensureDesignSystemDefaults,
+    withDesignSystemDefaults,
+} from "../../design-system";
+import { clamp, memoize } from "lodash-es";
+import { findClosestSwatchIndex, palette, Palette, PaletteType, Swatch } from "./palette";
+import {
+    ColorRecipe,
+    FillSwatch,
+    StatefulSwatchToColorRecipeFactory,
+    SwatchStates,
+} from "./common";
+
+/**
+ * Deltas to derive state swatches from the background
+ * All states have the same delta.
+ */
+export const neutralFillInputDelta: number = 4;
+
+/**
+ * Algorithm for determining neutral backplate colors
+ */
+const neutralFillInputAlgorithm: DesignSystemResolver<FillSwatch> = memoize(
+    (designSystem: DesignSystem): FillSwatch => {
+        const neutralPalette: Palette = palette(PaletteType.neutral)(designSystem);
+        const backgroundIndex: number = findClosestSwatchIndex(
+            PaletteType.neutral,
+            designSystem.backgroundColor
+        )(designSystem);
+        const fillIndex: number = clamp(
+            backgroundIndex - neutralFillInputDelta,
+            0,
+            neutralPalette.length - 1
+        );
+
+        return {
+            rest: neutralPalette[fillIndex],
+            hover: neutralPalette[fillIndex],
+            active: neutralPalette[fillIndex],
+            selected: neutralPalette[fillIndex],
+        };
+    },
+    (designSystem: DesignSystem): string => {
+        return designSystem.backgroundColor;
+    }
+);
+
+export function neutralFillInput(designSystem: DesignSystem): FillSwatch;
+export function neutralFillInput(
+    backgroundResolver: (designSystem: DesignSystem) => Swatch
+): (designSystem: DesignSystem) => FillSwatch;
+export function neutralFillInput(arg: any): any {
```

Comment:
total nit: don't we usually stay away from abbreviations (i.e. `arg`)

Reply:
I think "arg" is an industry-standard acronym - which is our guideline. Happy to change if folks don't think it is though.